language_server/src/main.cpp

```diff
@@ -82,10 +84,12 @@ class main_program final : public json_sink,
         , router(&lsp_queue)
         , dap_sessions(*this, json_output, &dap_telemetry_broker)
         , virtual_files(*ws_mngr, json_output)
+        , config_diags_provider(*ws_mngr)
     {
         router.register_route(dap_sessions.get_filtering_predicate(), dap_sessions);
         router.register_route(virtual_files.get_filtering_predicate(), virtual_files);
         router.register_route(external_files.get_filtering_predicate(), external_files);
+        router.register_route(config_diags_provider.get_filtering_predicate(), config_diags_provider);
```

Comment:
Is there a particular reason for going this router? The processing in WSmgr is now fully interruptible and it is possible to "jump" the queue there or execute synchronously without queueing.

Reply:
No, no particular reason. I'll take a look.